docs/router/ssr.md

```diff
@@ -73,4 +73,6 @@ const renderToStringWithData = async ({ location }) => {
 };
 ```
 
-Notice that we do not need to provide any `resourceData` object to the `ServerApp`,the `StaticRouter` handles this for us internally.
+Notice that we do not need to provide any `resourceData` object to the `ServerApp`, the `StaticRouter` handles this for us internally.
+
+To prevent slow APIs from causing long renders on the server you can optionally pass in `timeout` as an option to `StaticRouter.requestResources`. If a route resource does not return within the specified time then its data and promise will be set to null.
```

Comment:
We should add somewhere (in the `useResource` docs) a table about combinations of states:

| state | data | loading | error | promise |
|------|------|--------|------|------------|
| idle | `null` | `false` | `null` | `null` |
| loading | `null` or `{}` | `true` | `null` or `Error` | `Promise` |
| loading after ssr timeout | `null` | `true` | `TimeoutError` | `null` |
| fetch successful  | `{}` | `false` | `null` | `Promise` |
| fetch error   | `null` or `{}` | `false` | `Error` | `Promise` |

Also, we should add an explicit note in `useResource` docs saying that in SSR, `loading` can be `true` even when there is an error, but promise will be `null` because there is no `Suspense` support. So devs should give priority to `loading` to decide between loading or error state for their components, and only throw promise/error on the client